Cabal/Distribution/Simple/Utils/Json.hs

```diff
@@ -0,0 +1,34 @@
+module Distribution.Simple.Utils.Json
```

Comment:
I think this file should be removed, is that still true?

Reply:
Yes, definitely! We already have a JSON library (which is more or less a derivative of http://hackage.haskell.org/package/microaeson-0.1.0.0/docs/Data-Aeson-Micro.html) inside `exe:cabal`.

By moving show-build-info into `exe:cabal` we'd kill two birds with one stone...

Reply:
At this point I don't care whether we address this as part of this PR, or at a later stage; so feel free to consider this issue of the JSON library duplication punted to a follow-up PR

It may be worth leaving a TODO source-comment at the top of this module